### DISCUSSION TITLE
Import Always Has Today's Date

### DISCUSSION THREAD

**Discussion by darthrater78** [firefly-iii/firefly-iii#8329] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [ ] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [ ] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

I cannot get my imports to use anything but today's date.  Version 6.1.1

### Debug information

{
    "version": 3,
    "source": "ff3-importer-1.4.0",
    "created_at": "2024-01-02T21:35:48+01:00",
    "date": "j\/d\/Y",
    "default_account": 14,
    "delimiter": "comma",
    "headers": true,
    "rules": true,
    "skip_form": false,
    "add_import_tag": true,
    "roles": [
        "date_transaction",
        "_ignore",
        "description",
        "category-name",
        "amount",
        "note"
    ],
    "do_mapping": [
        false,
        false,
        false,
        false,
        false,
        false
    ],
    "mapping": [],
    "duplicate_detection_method": "classic",
    "ignore_duplicate_lines": true,
    "unique_column_index": 0,
    "unique_column_type": "internal_reference",
    "flow": "file",
    "content_type": "csv",
    "custom_tag": "",
    "identifier": "0",
    "connection": "0",
    "ignore_spectre_categories": false,
    "grouped_transaction_handling": "",
    "use_entire_opposing_address": false,
    "map_all_data": false,
    "accounts": [],
    "date_range": "",
    "date_range_number": 30,
    "date_range_unit": "d",
    "date_not_before": "",
    "date_not_after": "",
    "nordigen_country": "",
    "nordigen_bank": "",
    "nordigen_requisitions": [],
    "nordigen_max_days": "90",
    "conversion": false,
    "ignore_duplicate_transactions": true
}

### Expected behaviour

I noticed that my imports were using the date they were imported, and not the date of the transactions. I checked the docs, and googled the issue but could not get it to resolve. 

My bank uses m-d-y, so I updated the date format to m/d/Y, and then I noticed that there are no leading zeros for the month, so I used j/d/Y with the same result. 

![image](https://github.com/firefly-iii/firefly-iii/assets/94141126/fdaf325a-c78a-4e6c-9f0b-bcaa2c1f3da6)
![image](https://github.com/firefly-iii/firefly-iii/assets/94141126/7fb966b6-3b5a-4b60-90d0-def85656d9d3)

**Comment by JC5** [comment]
Can you share a row using a text editor? Excel parses the date for you.

**Reply by darthrater78** [reply]
![image](https://github.com/firefly-iii/firefly-iii/assets/94141126/28117971-9981-4861-be54-90953e5ea1d5)

    "version": 3,
    "source": "ff3-importer-1.4.0",
    "created": "2024-01-02T23:45:21+01:00",
    "date": "Y\/m\/d",
    
    I changed the date format to match and re-imported, same issue.

**Reply by darthrater78** [reply]
Edit: i was using / when I should have been using - as well, to match the source format. Looks like that fixed it, thank you!
